HousingRepairsOnline.AddressIngestion/Services/InsertAddressesToCosmosDb.cs

```diff
@@ -0,0 +1,46 @@
+namespace HousingRepairsOnline.AddressIngestion.Services
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using HACT.Dtos;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Extensions.Logging;
+
+    public class InsertAddressesToCosmosDb
+    {
+        private readonly IAsyncCollector<PropertyAddress> propertyAddressesOut;
+        private readonly ILogger logger;
+
+        public InsertAddressesToCosmosDb(IAsyncCollector<PropertyAddress> propertyAddressesOut, ILogger logger)
+        {
+            this.propertyAddressesOut = propertyAddressesOut;
+            this.logger = logger;
+        }
+
+        public async Task Execute(IEnumerable<PropertyAddress> propertyAddresses)
+        {
+            foreach (var propertyAddress in propertyAddresses)
+            {
+                if (propertyAddress.AddressLine.FirstOrDefault() is "" or null)
```

Comment:
It would be best to output all data validation errors and not just one at a time as has been done here.

Convert the `if-else` chaining to multiple `if`s that log the issue and set an invalid data flag to `true`, which is checked prior to adding the property address.
i.e.
```
bool isValidData = true;
if (<check 1>){
  logger.LogInformation...
  isValidData = false;
}
if (<check 2>){...}
if (<check 3>){...}

if (isValidData) {
  <add property>
}
```

Reply:
ahh yes, good point!